### DISCUSSION TITLE
[SOLVED] How to use Blazorise input controls in my custom razor components?

### DISCUSSION THREAD

**Discussion by gac-noob** [Megabit/Blazorise#1866] [Q&A] [answered]
Hi,

I am currently using Blazorise for validation and input controls (_see sample code below_):
```
<Validation>
     <Field>
          <FieldLabel>Reference No.</FieldLabel>
          <TextEdit Placeholder="put document serial number here" @bind-Text="@Document.RefNo">
               <Feedback>
                    <ValidationError />
               </Feedback>
          </TextEdit>
     </Field>
</Validation>
```

And I am wondering about two things:
     **1. I'd like to create my own custom razor components so that my code will be shorter and more readable.**

     `<MyTextEdit Title="Reference No." Placeholder="put document serial number here" @bind-Text="@Document.RefNo"/>`

My razor code looks like this:

```
<Validation>
     <Field>
          <FieldLabel>@Title</FieldLabel>
               <TextEdit Placeholder="@PlaceholderText" @bind-Text="@Document.RefNo">
               <Feedback>
                    <ValidationError />
               </Feedback>
          </TextEdit>
     </Field>
</Validation>

@code {
    [Parameter] public string Title { get; set; } = "";
    [Parameter] public string PlaceholderText { get; set; } = "";
    
	[Parameter] private string _datasource { get; set; } = "";
	[Parameter] public string DataSource
        {
            get => _datasource;
            set
            {
                if (_datasource == value) return;
                _datasource = value;
                DataSourceChanged.InvokeAsync(value);
            }
        }
	[Parameter] public EventCallback<string> DataSource { get; set; }
}
```

The problem is this does not work, the validation does not get called, so no validation error is displayed..which defeats the purpose of creating the custom razor component in the first place..hahaha

Can anybody please teach me how to do this properly? If this is not advisable, please let me know also, I'm here to learn! :)

**2. Is it going to make my app perform slower because I am nesting razor components?**

Thank you in advance!

**Comment by stsrki** [comment, marked as answer]
Check this answer https://github.com/stsrki/Blazorise/discussions/1722

**Reply by gac-noob** [reply]
Thank you very much for pointing me at the solution sir Mladen, everything works perfectly now!!! :)
